spec/ravelin/proxy_client_spec.rb

```diff
@@ -0,0 +1,450 @@
+      it 'calls #get with customer id' do
+        allow(Ravelin::Tag).to receive(:new) { tag }
+        expect(client).to receive(:get).with('/ravelinproxy/v2/tag/customer/123')
+        client.get_tag
+      end
+    end
+  end
+
+  describe '#post' do
+    let(:client) { described_class.new(base_url: base_url, username: username, password: password) }
+    let(:event) do
+      double('event', name: 'ping', serializable_hash: { name: 'value' })
+    end
+
+    let(:headers) do
+      {
+        'Authorization' =>"Basic #{base64_enc_user_pass}",
+        'User-Agent'    => "Ravelin Proxy RubyGem/#{Ravelin::VERSION}".freeze
+      }
+    end
+
+    before do
+      allow(Ravelin::Event).to receive(:new).and_return(event)
+    end
+
+    it 'calls Ravelin Proxy with correct headers and body' do
+      stub = stub_request(:post, "http://127.0.0.1:8020/ravelinproxy/v2/ping").
+        with(
+          body: { name: 'value' }.to_json,
+          headers: headers
+        ).and_return(
+          headers: { 'Content-Type' => 'application/json' },
+          body: '{}')
+      client.send_event
+
+      expect(stub).to have_been_requested
+    end
+
+    context 'response' do
+      before do
+        stub_request(:post, 'http://127.0.0.1:8020/ravelinproxy/v2/ping').
+          to_return(
+            status: response_status,
+            body: body
+          )
+      end
+
+      context 'successful' do
+        shared_examples 'successful request' do
+          it 'returns the response' do
+            expect(client.send_event).to be_a(Ravelin::Response)
+          end
+
+          it "not treated as an error" do
+            expect(client).to_not receive(:handle_error_response)
+
+            client.send_event
+          end
+        end
+
+        context 'when the response code is 200' do
+          let(:response_status) { 200 }
+          let(:body) { '{}' }
+          it_behaves_like 'successful request'
+        end
+
+        context 'when the response code is 200' do
+          let(:response_status) { 204 }
+          let(:body) { '' }
+          it_behaves_like 'successful request'
+        end
+      end
+
+      context 'error' do
+        let(:response_status) { 400 }
+        let(:body) { '{}' }
+        it 'handles error response' do
+          expect(client).to receive(:handle_error_response).
+            with(kind_of(Faraday::Response))
+
+          client.send_event
+        end
+      end
+    end
+  end
+
+  describe '#delete' do
+    let(:client) { described_class.new(base_url: base_url, username: username, password: password) }
+    let(:tag) do
+      double('tag', name: 'ping', serializable_hash: { "customerId" => '123', "tagNames" => ['foo', 'bar'] })
+    end
+
+    before do
+      allow(Ravelin::Tag).to receive(:new).and_return(tag)
+    end
+
+    it 'calls Ravelin with correct headers and body' do
+      stub = stub_request(:delete, 'http://127.0.0.1:8020/ravelinproxy/v2/tag/customer?customerId=123&tagName=foo,bar').
```

Comment:
Nit: I see the url `http://127.0.0.1:8020` repeated a lot, there may be an opportunity to tidy it with a variable.  This is just a passing observation, I'm not passionate about it.

Reply:
agreed. I'll see what I can do.

There is no place like `127.0.0.1`

Reply:
https://media.giphy.com/media/tnYri4n2Frnig/giphy.gif

Reply:
![image](https://user-images.githubusercontent.com/42939735/81080512-b5b02c00-8ee8-11ea-8dc5-fa4a445f75a3.png)

Reply:
![image](https://user-images.githubusercontent.com/42939735/81080567-c5c80b80-8ee8-11ea-9c62-456f594c4a57.png)

Reply:
fixed in 8823f8377a809681cb0588b34088e146c72bc49e